onebusaway-android/src/main/java/org/onebusaway/android/ui/PreferencesActivity.java

```diff
@@ -366,10 +380,52 @@ public boolean onPreferenceClick(Preference pref) {
             // Try to push firebase data to the server
             FirebaseDataPusher pusher = new FirebaseDataPusher();
             pusher.push(this);
+        } else if (pref.equals(mVolumePref)) {
+            reminderVolumeDialog();
         }
         return true;
     }
 
+    /*
+    * Dialog to set the volume for reminder using seekbar
+    * save volume in shared Preferences
+    */
+    private void reminderVolumeDialog() {
+        final int progress = settings.getInt("reminder_volume",100);
+        LayoutInflater inflater = getLayoutInflater();
+        View customLayout = inflater.inflate(R.layout.reminder_volume_adjuster, null);
+        SeekBar seekBar = customLayout.findViewById(R.id.seekBar);
+        AlertDialog.Builder builder = new AlertDialog.Builder(this)
+                .setView(customLayout)
+                .setPositiveButton(R.string.save,
+                        (dialog, which)->{
+                                setReminderVolume(seekBar.getProgress());
+                        })
+                .setNegativeButton(R.string.cancel,
+                        (dialog, which)->{
+                            dialog.dismiss();
+                        }
+                );
+        seekBar.setProgress(progress);
+        AlertDialog dialog = builder.create();
+        dialog.show();
+    }
+
+    private void setReminderVolume(int progress) {
+        AudioManager audioManager = null;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
```

Comment:
if this feature only works on Android M and above, then it should only be displayed to users on Android M and above.